Code/GraphMol/RGroupDecomposition/RGroupData.h

```diff
@@ -57,9 +57,7 @@ struct RGroupData {
 
     mols.push_back(newMol);
     std::string smi = MolToSmiles(*newMol, true);
-    // REVIEW: we probably shouldn't be using a set here... the merging of
-    // duplicates is likely not what we want
-    smilesSet.insert(smi);
+    smilesVect.emplace_back(smi);
```

Comment:
Does Emplace back really doesn't help here?   I'm not sure if the optimizer reduces this to:

```
smilesVect.emplace_back(MolToSmiles(*newMol, true));
```

which would use copy elision.

I tend to use emplace_back when I don't want to use a constructor or their is no optimized copy constructor.

```
vect.emplace_backI("asdfasdf");
```

as opposed to

```
vect.push_back(string("asdkjfasdf"));
```

Reply:
Yeah, it should have rather been `push(back(std::move(smi))`. but there's no need to store in `smi` altogether.